src/components/carousel/index.js

```diff
@@ -493,28 +535,26 @@ export default class Carousel extends BaseComponent {
   }
 }
 
-function createStyles({containerPaddingVertical = 0}) {
-  return StyleSheet.create({
-    counter: {
-      paddingHorizontal: 8,
-      paddingVertical: 3, // height: 24,
-      borderRadius: 20,
-      backgroundColor: Colors.rgba(Colors.black, 0.6),
-      position: 'absolute',
-      top: 12,
-      right: 12
-    },
-    pageControlContainerStyle: {
-      position: 'absolute',
-      bottom: 16,
-      alignSelf: 'center'
-    },
-    pageControlContainerStyleUnder: {
-      marginVertical: 16 - containerPaddingVertical
-    },
-    hiddenText: {
-      position: 'absolute', 
-      width: 1
-    }
-  });
-}
+export {Carousel}; // For tests
+export default asBaseComponent(forwardRef(Carousel));
```

Comment:
We don't want to forward ref in the case of the Carousel. 
Forwarding ref means the user will have access to the internal ScrollView ref instead of the Carousel component.
In our case the Carousel provide imperative methods to use with ref like `goToPage`

Anyway, even if we would have want to forward ref, notice that you didn't really forward it completely. 
See how it's done in the View component 
https://github.com/wix/react-native-ui-lib/blob/master/src/components/view/index.tsx#L117
You still need to pass the `forwardedRef` injected prop

Reply:
Ok got it. Thanks